File: shoe_wash_app/backend/src/domain/pickup_request.py
from dataclasses import dataclass
from datetime import date, datetime
from typing import Optional


class PickupRequestStatus:
    REQUESTED = "requested"
    CONFIRMED = "confirmed"
    COLLECTED = "collected"
    CANCELLED = "cancelled"


@dataclass
class PickupRequest:
    """A customer's request for the shop to come collect their items --
    exists before any Load does, since the shop doesn't know item class,
    quantity, or price until the items are actually in hand."""

    id: Optional[int]
    customer_name: str
    phone: str
    address: str
    requested_at: datetime
    status: str = PickupRequestStatus.REQUESTED
    scheduled_date: Optional[date] = None
    notes: Optional[str] = None
    collected_load_id: Optional[int] = None

File: shoe_wash_app/mobile_app/lib/models/pickup_request.dart
class PickupRequest {
  final int id;
  final String customerName;
  final String phone;
  final String address;
  final String requestedAt;
  final String status;
  final String? scheduledDate;
  final String? notes;
  final int? collectedLoadId;

  PickupRequest({
    required this.id, required this.customerName, required this.phone, required this.address,
    required this.requestedAt, required this.status, this.scheduledDate, this.notes, this.collectedLoadId,
  });

  factory PickupRequest.fromJson(Map<String, dynamic> json) {
    return PickupRequest(
      id: json['id'], customerName: json['customer_name'], phone: json['phone'], address: json['address'],
      requestedAt: json['requested_at'], status: json['status'], scheduledDate: json['scheduled_date'],
      notes: json['notes'], collectedLoadId: json['collected_load_id'],
    );
  }
}